### PR TITLE
build(dependencies): updated typescript, ts-node, added @types/chai

### PR DESCRIPTION
**Pull request by nchursin**
Split dependency updates from https://github.com/serenity-js/serenity-js/pull/911.

### CODE REVIEW

File: documentation/esdoc-external-docs-plugin/package.json [hunk 1]
```diff
@@ -34,9 +34,12 @@
     "npm": "^6 || ^7"
   },
   "dependencies": {
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline documentation/esdoc-external-docs-plugin/package.json:37]: Maybe we could move them to `devDependencies` instead?

File: documentation/esdoc-template/package.json [hunk 2]
```diff
@@ -32,5 +32,10 @@
   "engines": {
     "node": "^12 || ^14 || ^16",
     "npm": "^6 || ^7"
+  },
+  "dependencies": {
```
Comment (jan-molak) [inline documentation/esdoc-template/package.json:36]: `dependencies` -> `devDependencies`?

File: documentation/website/package.json [hunk 3]
```diff
@@ -71,5 +71,10 @@
     "minimatch": "^3.0.4",
     "node-sass": "^6.0.1",
     "yaml": "^1.10.2"
+  },
```
Comment (jan-molak) [inline documentation/website/package.json:74]: `dependencies` -> `devDependencies`

File: examples/calculator-app/package.json [hunk 4]
```diff
@@ -45,10 +45,13 @@
     "ts-node": "^10.1.0"
   },
   "dependencies": {
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline examples/calculator-app/package.json:48]: `dependencies` -> `devDependencies`

File: integration/cucumber-1-runner/package.json [hunk 5]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "1.3.3"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-1-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber-2-runner/package.json [hunk 6]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "2.3.1"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-2-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber-3-runner/package.json [hunk 7]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "3.2.1"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-3-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber-4-runner/package.json [hunk 8]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "4.2.1"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-4-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber-5-runner/package.json [hunk 9]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "5.1.0"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-5-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber-6-runner/package.json [hunk 10]
```diff
@@ -39,7 +39,10 @@
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
-    "cucumber": "6.0.5"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/cucumber-6-runner/package.json:42]: `dependencies` -> `devDependencies`

File: integration/cucumber/package.json [hunk 11]
```diff
@@ -38,7 +38,10 @@
     "@integration/cucumber-6-runner": "2.0.0",
     "@integration/testing-tools": "2.0.0",
     "@serenity-js/core": "^2.0.0",
-    "@serenity-js/cucumber": "^2.0.0"
+    "@serenity-js/cucumber": "^2.0.0",
```
Comment (jan-molak) [inline integration/cucumber/package.json:41]: `dependencies` -> `devDependencies`

File: integration/protractor-cucumber/package.json [hunk 12]
```diff
@@ -35,7 +35,10 @@
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/cucumber": "^2.0.0",
     "@serenity-js/protractor": "^2.0.0",
-    "protractor": "^7.0.0"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/protractor-cucumber/package.json:38]: `dependencies` -> `devDependencies`

File: integration/protractor-mocha/package.json [hunk 13]
```diff
@@ -34,7 +34,10 @@
     "@serenity-js/core": "^2.0.0",
     "@serenity-js/mocha": "^2.0.0",
     "@serenity-js/protractor": "^2.0.0",
-    "protractor": "^7.0.0"
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/protractor-mocha/package.json:37]: `dependencies` -> `devDependencies`

File: integration/testing-tools/package.json [hunk 14]
```diff
@@ -42,8 +42,11 @@
     "@serenity-js/core": "^2.0.0"
   },
   "dependencies": {
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline integration/testing-tools/package.json:45]: `dependencies` -> `devDependencies`

File: packages/core/package.json [hunk 15]
```diff
@@ -34,6 +34,7 @@
     "site": "esdoc -c .esdoc.js"
   },
   "dependencies": {
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline packages/core/package.json:37]: `dependencies` -> `devDependencies`

File: packages/cucumber/package.json [hunk 16]
```diff
@@ -53,7 +53,10 @@
   },
   "peerDependencies": {
     "@cucumber/cucumber": "^7.3.0",
-    "cucumber": "^1.3.3 || ^2.3.1 || ^3.2.1 || ^4.2.1 || ^5.0.0 || ^6.0.0"
+    "cucumber": "^1.3.3 || ^2.3.1 || ^3.2.1 || ^4.2.1 || ^5.0.0 || ^6.0.0",
```
Comment (jan-molak) [inline packages/cucumber/package.json:56]: `peerDependencies` don't need changing I think; Those are modules require by a user of Serenity/JS to have.

File: packages/protractor/package.json [hunk 17]
```diff
@@ -61,7 +61,10 @@
     "@serenity-js/jasmine": "^2.29.0",
     "@serenity-js/mocha": "^2.29.0",
     "protractor": "^5.0.0 || ^7.0.0",
-    "selenium-webdriver": "^3.6.0"
+    "selenium-webdriver": "^3.6.0",
+    "@types/chai": "^4.2.21",
```
Comment (jan-molak) [inline packages/protractor/package.json:65]: I don't think those need to be listed as `peerDependencies`, `devDependencies` should be enough

**Review by jan-molak — CHANGES_REQUESTED** [review]
Thanks @nchursin! Could we please ensure that `@types/chai`, `ts-node` and `typescript` are listed as `devDependencies` rather than `dependencies`? I'd also avoid changing the `peerDependencies` section, since it describes modules Serenity/JS requires its consumers to have.

File: documentation/website/package.json (hunk 3 above)
Comment (nchursin) [reply documentation/website/package.json:74]: Argh, nuts. I only fixed it for files that had conflicts. Will fix

**Comment by nchursin** [comment]
Done. I actually didn't have the intention to change peer dependencies, but when you look through so many `package.json` files, you may just confuse `peer` and `dev` :)

**Pull request merged**